File: app/src/main/java/com/worldventures/dreamtrips/modules/config/model/Configuration.java
package com.worldventures.dreamtrips.modules.config.model;

public class Configuration {

   private final UpdateRequirement updateRequirement;
   private final VideoRequirement videoRequirement;

   public Configuration() {
      videoRequirement = new VideoRequirement();
      updateRequirement = new UpdateRequirement();
   }

   public Configuration(UpdateRequirement updateRequirement, VideoRequirement videoRequirement) {
      this.updateRequirement = updateRequirement;
      this.videoRequirement = videoRequirement;
   }

   public UpdateRequirement getUpdateRequirement() {
      return updateRequirement;
   }

   public VideoRequirement getVideoRequirement() {
      return videoRequirement;
   }
}

File: app/src/main/java/com/worldventures/dreamtrips/modules/config/model/Configuration.kt
package com.worldventures.dreamtrips.modules.config.model

import com.esotericsoftware.kryo.DefaultSerializer
import com.esotericsoftware.kryo.serializers.CompatibleFieldSerializer

@DefaultSerializer(CompatibleFieldSerializer::class)
data class Configuration(val updateRequirement: UpdateRequirement? = null,
                         val videoRequirement: VideoRequirement = VideoRequirement(),
                         val travelBannerRequirement: TravelBannerRequirement? = null)
